packages/cli/src/util/dev/parse-query-string.ts

```diff
@@ -38,9 +38,9 @@ export function parseQueryString(
  */
 export function formatQueryString(
   query: Record<string, string[]> | undefined
-): string | undefined {
+): string | null {
```

Comment:
How come this change was necessary?

Reply:
The URL property the result of this function was being assigned too requires string | null instead of undefined

Reply:
This seems like a bug. We treat null and undefined differently in this function

Reply:
I believe its a bug with this function. All usage of `formatQueryString` is for assigning a value to a `url.UrlWithStringQuery` instance `search` property. Which is typed as `string | null`, not `string | undefined`. The easiest (and what I believe the correct change here), is that `formatQueryString` should be changed to return `string | null`. 
 
<img width="455" alt="Screen Shot 2022-11-04 at 12 13 16 PM" src="https://user-images.githubusercontent.com/16144158/200046578-74495095-ba19-404d-8f69-96e241b18a77.png">

Furthermore, the `parseQueryString` function is always called with properties typed as `string | null`, hence that change. That function already has logic checking the truthiness of the parameter so it was an easy change.

<img width="456" alt="Screen Shot 2022-11-04 at 12 15 15 PM" src="https://user-images.githubusercontent.com/16144158/200046927-62fcec5c-ecd6-470b-a97c-8478467072bc.png">

Reply:
Another solution is to add `?? null` to all of the `formatQueryString` calls i.e. `origUrl.search = formatQueryString(origQuery) ?? null` and `?? undefined` to all of the parseQueryString calls i.e. `const origQuery = parseQueryString(origUrl.search ?? undefined)`

Reply:
At a higher level, this function could, and possibly should, always return a `string`. If there's no `query`, then the result would simply be `""`. Then the body of the function could then be simplified down to:

```
return new URLSearchParams(query).toString();
```